lib/app/models/testing_instance.rb

```diff
@@ -281,6 +275,40 @@ def token_expiration_time
         token_retrieved_at + token_expires_in.seconds
       end
 
+      def get_sequence_requirement(requirement)
+        return unless requirement&.dig(:name)
+
+        attributes = { name: requirement[:name],
+                       testing_instance: self,
+                       label: requirement[:name],
+                       value: '' }
+          .merge(requirement)
+
+        condition_attributes = [:name, :testing_instance]
+
+        # Get the conditions that determine whether the record already exists
+        conditions = attributes.select { |key, _value| condition_attributes.include?(key) }
+
+        SequenceRequirement.first_or_create(conditions, attributes)
```

Comment:
I think lines 287-290 are overcomplicating things.
```suggestion
        SequenceRequirement.first_or_create({ name: requirement[:name], testing_instance: self }, attributes)
```